mitmproxy/net/http/request.py

```diff
@@ -367,13 +367,64 @@ def _get_urlencoded_form(self):
                 pass
         return ()
 
+    def _sanitized_form_data_replacement(self, form_data):
```

Comment:
> If this is what you mean, I fully agree

Yes. The motivation is not necessarily to use this elsewhere, the motivation is to keep encode side-effect free to simplify testing.